### PR TITLE
Make generateapis.sh aware of SYNTHTOOL_GOOGLEAPIS

### LINKED ISSUE #3339: Doesn't support env var --> SYNTHTOOL_GOOGLEAPIS (local googleapis)
**Is your feature request related to a problem? Please describe.**
It doesn't support $SYNTHTOOL_GOOGLEAPIS (env var that points to local googleapis/googleapis).  Other languages supports $SYNTHTOOL_GOOGLEAPIS, and I thnk it might be good idea to add this as option and keep it consistent throughout all languages.

 **Describe the solution you'd like**
Add an option to read $SYNTHTOOL_GOOGLEAPIS (points to local googleapis) if that env variable exists.

### PR DESCRIPTION
**Pull request by jskeet**
Expectations:
- SYNTHTOOL_GOOLEAPIS is either a path relative to the google-cloud-dotnet root, or an absolute path, without spaces
- Other generator code (e.g. pre/post-build fix-ups) doesn't need to find the protos

The second expectation is known to be violated in the Google.LongRunning package, but that's probably okay for now.

This does as much as we're likely to want to do to fix #3339.

### CODE REVIEW

File: generateapis.sh [hunk 1]
```diff
@@ -40,21 +46,26 @@ fetch_github_repos() {
     git clone --recursive https://github.com/googleapis/gapic-generator-csharp
   fi
 
-  if [ -d "googleapis" ]
+  if [[ "$SYNTHTOOL_GOOGLEAPIS" != "" ]]
```
Comment (chrisdunelm) [inline generateapis.sh]: I'm not very familiar with the context of the PR, but I was expecting this to simply replace "googleapis" with $GOOGLEAPIS ?

Comment (chrisdunelm) [reply generateapis.sh]: OK, so shouldn't it be `==`, not `!=`?

And what's going on - you're meant to be on vacation??

Comment (jskeet) [reply generateapis.sh]: Was just idly checking emails, but won't look at detail of PR now :)

Comment (jskeet) [reply generateapis.sh]: Whoops - yes, that should indeed be ==. Fixed, and removed the echo lower down.

**Pull request merged**